### ISSUE TITLE
showimage function: width and height attribute

### ISSUE THREAD

**Issue by nlemoine** [bolt/bolt#2275]
When using the `showimage` template tag. The image attributes used are set to the size passed to the function. For example, I want a maximum width of 800 pixels:
```
record.image.filename|showimage(800, 9999, 'r')
```
Will output:
```
<img src="http://path/to/image.jpg" width="800" height="9999" />
```

As the "r" filter resizes given a maximum width or height, the other attribute should be set accordingly, preserving original aspect ratio. If I have a 1600x700 image with the same parameters as the example above, the actual function will ouput:
```
<img src="http://path/to/image.jpg" width="800" height="9999" />
```
Whereas it should output:
```
<img src="http://path/to/image.jpg" width="800" height="350" />
```
<hr />
I also noticed a little mistake in the 2.0 docs:

![capture d ecran 2014-12-05 a 15 13 27](https://cloud.githubusercontent.com/assets/2526939/5316895/513b8e46-7c91-11e4-9e66-0ddcf7009a01.png)

You can't pass an image array directly, you have to pass the filename:
```
{{ record.photo.filename|showimage(800, 600) }}
```
It would actually be nice to pass the image array instead of filename. Or both.

**Comment by SahAssar** [comment]
Just commenting here since I just opened a PR regarding this:

To get proper height/width variables in a showimage has been supported for some time it seems, but only when calling with a filename. For example `record.image.filename|showimage(800, 9999, 'r')` Would work if called like this instead even before the "fix": `record.image.filename|showimage(800, 0, 'r')`. What the fix changes is that you can call it with an image from an imagelist directly (since it checks for an array), or with `record.values.image|showimage(800, 0, 'r')` if you want it to output a proper alt tag.

**Comment by bobdenotter** [comment]
Closed by #4231
